spec/API_specification/creation_functions.md

```diff
@@ -322,7 +322,7 @@ Returns coordinate matrices from coordinate vectors.
 
 -    **arrays**: _&lt;array&gt;_
 
-     -   an arbitrary number of one-dimensional arrays representing grid coordinates. Must have numeric data types.
+     -   an arbitrary number of one-dimensional arrays representing grid coordinates. Each array must have the same numeric data type.
```

Comment:
We should use "should" here instead of "must". That way libraries can support mixed dtype behavior if they want, but the spec only specifies behavior when they match.

Reply:
Makes sense, thanks!

I also changed the must to should in the modified return part of meshgrid. I *think* it's appropriate too?

```diff
- Each returned array must have the same data type as the input arrays.
+ Each returned array should have the same data type as the input arrays.
```

Reply:
I think so, since any sort of type promotion would only occur when the inputs have different dtypes anyway, which is outside of the scope of the spec.